pages/board/[id].tsx

```diff
@@ -0,0 +1,71 @@
+import BoardComment from "../../components/BoardComment";
+import BoardCommentList from "../../components/BoardCommentList";
+import Header from "../../components/Header";
+import styles from "../../components/BoardDetail.module.css";
+import Link from "next/link";
+import back from "../../public/back.svg";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import BoardDetailInfo from "../../components/BoardDetailInfo";
+import { getArticlesComment, getArticlesId } from "../../lib/api";
+import { useEffect, useState } from "react";
+import { Article, Comment } from "../../components/type";
+
+interface board {
+  articleId: number;
+}
+interface comment {
+  articleId: number;
+  limit: number;
+}
+
+export default function BoardDetail() {
+  const router = useRouter();
+  const { id } = router.query;
+  const [article, setArticle] = useState<Article | null>(null);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+
+  async function getBoardList({ articleId }: board) {
+    try {
+      const response = await getArticlesId({ articleId });
+      setArticle(response);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  async function getCommemtList({ articleId, limit }: comment) {
+    try {
+      const response = await getArticlesComment({ articleId, limit });
+      setComments(response.list);
+    } catch (error) {
+      console.log("Error fetching articles:", error);
+      throw error;
+    }
+  }
+
+  useEffect(() => {
+    if (id) {
+      getBoardList({ articleId: parseInt(id as string) });
+      getCommemtList({ articleId: parseInt(id as string), limit: 3 });
+    }
```

Comment:
id의 타입이 배열 또는 문자열이여서 타입단언을 사용하신거라면
아예, if 조건문에서 부터 타입을 좁혀주면 더 간단히 작성할 수 있습니다.
또한, id는 가능성이 매우 낮지만 url 입력값에 따라 배열로 들어올 수 있는 값이기도 해서
배열로 들어올 경우를 한 번 필터링해주는 효과도 있습니다. 😁

```suggestion
    if (id && !Array.isArray(id)) {
      getBoardList({ articleId: parseInt(id) });
      getCommemtList({ articleId: parseInt(id), limit: 3 });
    }
```